wasm-runtime/polkadot/src/codec/slicable.rs

```diff
@@ -77,8 +79,16 @@ impl Slicable for Vec<u8> {
 	fn from_slice(value: &[u8]) -> Option<Self> {
 		Some(value[4..].to_vec())
 	}
-	fn set_as_slice<F: FnOnce(&mut [u8]) -> bool>(_fill_slice: F) -> Option<Self> {
-		unimplemented!();
+	fn set_as_slice<F: Fn(&mut [u8], usize) -> bool>(fill_slice: &F) -> Option<Self> {
+		u32::set_as_slice(fill_slice).and_then(|len| {
+			let mut v = Vec::with_capacity(len as usize);
+			unsafe { v.set_len(len as usize); }
+			if fill_slice(&mut v, 4) {
```

Comment:
better to just zero the vector:
  1. it's 4 bytes
  2. `fill_slice` might cause undefined behavior
  3. unsafe code is bad
  4. is this a bottleneck?

Reply:
it's not 4 bytes. the `4` is the four byte offset; the vector could be huge e.g. a multi-MB block.